Support publishing using Ivy (#222)
gradle-nexus/publish-plugin: README.md

    sign(publishing.publications["mavenJava"])
}
```

#### Publishing with Ivy ####

There are cases where it may be necessary to use the `ivy-publish` plugin instead of `maven-publish`.
For example, when publishing Sbt plugins the directory structure needs to be customized which is only possible with Gradle's `IvyArtifactRepository`.

In such cases, you need to apply the `ivy-publish` plugin and configure the `nexusPublishing` extension's `publicationType` to `IVY` (default is `MAVEN`).

In case of Ivy publishing, because of compatibility with Sonatype the nexus repository layout will be used by default

```groovy
nexusPublishing {
    publicationType = io.github.gradlenexus.publishplugin.NexusPublishExtension.PublicationType.IVY
}
```

Or use the kotlin DSL:

```kotlin
nexusPublishing {

>     publicationType.set(io.github.gradlenexus.publishplugin.NexusPublishExtension.PublicationType.IVY)

Reviewer: This needs to be updated now that `publicationType` is defined on `NexusRepository`

}
```

##### Using Ivy repositories with different artifact patterns ####

In case of ivy it's possible to override the default artifact pattern that is used, which is the Maven pattern due to compatibility reasons with sonatype

To change the pattern of artifacts and ivy files use

```groovy

> ivyPatternLayout {

Reviewer: I think we should show how to configure this on a repo here and below to make it clear where `ivyPatternLayout` needs to be called.

    sign(publishing.publications["mavenJava"])
}
```

> #### Publishing with ivy ####
> 
> Instead of the `maven-publish` plugin, the `ivy-publish` plugin can be used as well to publish to the nexus repository
> 
> In this case the different plugin must be applied and the `nexusPublishing` extension's publishingType property must be configured.
> 
> The default `publicationType` is `MAVEN`

Reviewer: ```suggestion
#### Publishing with Ivy ####

There are cases where it may be necessary to use the `ivy-publish` plugin instead of `maven-publish`.
For example, when publishing Sbt plugins the directory structure needs to be customized which is only possible with Gradle's `IvyArtifactRepository`.

In such cases, you need to apply the `ivy-publish` plugin and configure the `nexusPublishing` extension's `publicationType` to `IVY` (default is `MAVEN`).
```

There are cases where it may be necessary to use the `ivy-publish` plugin instead of `maven-publish`.
For example, when publishing Sbt plugins the directory structure needs to be customized which is only possible with Gradle's `IvyArtifactRepository`.

In such cases, you need to apply the `ivy-publish` plugin and configure the `publicationType` fore each `NexusRepository`, that should be ivy compatible, to `IVY` (default is `MAVEN`).

In case of Ivy publishing, because of compatibility with Sonatype the nexus repository layout will be used by default

```groovy
nexusPublishing {
    respositories {
        ivyRepository {
            publicationType = io.github.gradlenexus.publishplugin.NexusRepository.PublicationType.IVY
        }
    }
}
```

Or use the kotlin DSL:

```kotlin
nexusPublishing {
    respositories {

>         ivyRepository {

Reviewer: ❌ This will not work in Kotlin DSL, will it?

```suggestion
        register("ivyRepository") {
```

            publicationType.set(io.github.gradlenexus.publishplugin.NexusRepository.PublicationType.IVY)
        }
    }    
}
```

##### Using Ivy repositories with different artifact patterns ####

In case of ivy it's possible to override the default artifact pattern that is used, which is the Maven pattern due to compatibility reasons with sonatype

To change the pattern of artifacts and ivy files configure the `ivyPatternLayout` on each repository that should be used with this layout with:

```groovy
nexusPublishing {
    respositories {
        ivyRepository {
            ivyPatternLayout {
                artifact "[organisation]/[module]_foo/[revision]/[artifact]-[revision](-[classifier])(.[ext])"
                m2compatible = true
            }
        }
    }
}
```

Or use the kotlin DSL:

```kotlin
nexusPublishing {
    respositories {

>         ivyRepository {

Reviewer: Ditto.

```suggestion
        register("ivyRepository") {
```
